walkbox.go

```diff
@@ -79,3 +79,58 @@ func (w *WalkBox) ContainsPoint(p *Positionf) bool {
 
 	return numberOfIntersections%2 == 1 // Odd count means inside
 }
+
+// IsConnected checks if WalkBoxes are connected.
+func (w *WalkBox) IsConnected(otherWalkBox *WalkBox) bool {
+	for _, vertex := range otherWalkBox.vertices {
+		if w.ContainsPoint(vertex) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
This definition is only considering boxes connected if they overlap in one of their vertices. It is excluding boxes that are really adjacent. 

In the references I shared in https://github.com/apoloval/pctk/issues/19, it is mentioned that Scumm does it with a simple solution. Two boxes are adjacent as long as they share some horizontal or vertical edge. In ScummVM source code, we can find a very simple implementation of it.

Reply:
> This definition is only considering boxes connected if they overlap in one of their vertices. It is excluding boxes that are really adjacent.

Checking by vertex vs segments is more flexible and less restrictive. The effect is the opposite that you mention: we are including "non real adjacent" if we understand a "real adjacent" those boxes that  must shares one side of the polygon cause we only need a single point to be considered as neighbours not a segment (and no need 16 rotations to check it).


Taking a look to the scummvm implementation:

> // Roughly, the idea of this algorithm is to search for sides of the given
	// boxes that touch each other.
	// In order to keep the code simple, we only match the upper sides;
	// then, we "rotate" the box coordinates four times each, for a total
	// of 16 comparisons.



The difference here is, we are only checking one vertex is touching instead to check two to be sure there is no single point of contact but a segment (not needing 16 rotations for that)

I can't see cons in this moment for this implementation (other future functions to be implemented as ClosesPointToBox, ClosesPoint etc must work also with a single point of contact), only, we are not mimicking the original implementation but let me know, you have more context and maybe Im missing things.

Reply:
> I think the matrix should be an actual matrix. For a set of N boxes, an 2 dimensional array of NxN. Where each i, j coordinate indicates the next box to go if you want to move from box i to box j.

Related with  this, totally agree. We can remove the pseudo-graph implementation and use a matrix of neighbours to avoid DFS / BFS algorithms to obtain the shortest path

Gonna go with this change removing the `Add` function (moving this logic during the NewWalkBoxMatrix), Thanks!!

Reply:
Ok, now I understand. If two convex polygons overlap it is because one of them have one vertex inside the other. 

However, in this implementation you are only checking whether the vertices of _other_ are inside of _w_. But that's only one side of it. It could happen that a vertex of _w_ is inside _other_, as in:

![image](https://github.com/user-attachments/assets/1285a10e-65a5-4d11-99da-c9f6e0fcab60)

Reply:
I see, you are totally right! This is a great example
With the current implementation W is adjacent of Other, but Other is not adjacent of W.

Checking with segments doesnt solve the problem, now I see why the rotations 😅 
Gonna take a look to solve this (and add a unit test for this)

Thanks a lot for taking care of it! 🙇

Reply:
You are completely right for scenarios like this one 👇 
<img width="252" alt="image" src="https://github.com/user-attachments/assets/0aba5861-ea44-4217-b866-b6d44f7d842e">

Just added a two-way check adjacency as you suggested solves the issue.

Thanks a lot for taking care